src/runtime.js

```diff
@@ -7,19 +8,24 @@ export const runtime = {
         addListener: jest.fn(),
       },
       onMessage: {
-        addListener: jest.fn(),
+        addListener: jest.fn(listener => {
+          listeners.push(listener);
+        }),
       },
     };
     return connection;
   }),
   sendMessage: jest.fn((message, cb) => {
+    listeners.forEach(listener => listener(message));
     if (cb !== undefined) {
       return cb();
     }
     return Promise.resolve();
   }),
   onMessage: {
-    addListener: jest.fn(),
+    addListener: jest.fn(listener => {
+      listeners.push(listener);
+    }),
     removeListener: jest.fn(),
     hasListener: jest.fn(),
```

Comment:
Want to fill out the has and remove as well?

```suggestion
    hasListener: jest.fn(listener => listeners.includes(listener)),
```